// pocketbase-react/lib/context/auth.js
"use strict";

exports.__esModule = true;
exports.AuthProvider = exports.AuthContext = void 0;
var React = _interopRequireWildcard(require("react"));
var _useClientContext = require("../hooks/useClientContext");
var _Storage = require("../service/Storage");
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const AuthContext = (0, React.createContext)({});
exports.AuthContext = AuthContext;
const AuthProvider = props => {
  const client = (0, _useClientContext.useClientContext)();
  const [authProviders, setAuthProviders] = React.useState();
  const actions = {
    registerWithEmail: async (email, password) => {
      await (client == null ? void 0 : client.collection('users').create({
        email: email,
        password: password,
        passwordConfirm: password
      }));
    },
    signInWithEmail: async (email, password) => {
      await (client == null ? void 0 : client.collection('users').authWithPassword(email, password));
    },
    signInWithProvider: async provider => {
      const authProvider = authProviders == null ? void 0 : authProviders.find(p => p.name === provider);
      const redirectURL = typeof document !== 'undefined' ? props.webRedirectUrl : props.mobileRedirectUrl;
      const url = (authProvider == null ? void 0 : authProvider.authUrl) + redirectURL;
      await _Storage.StorageService.set('provider', JSON.stringify(authProviders));
      await props.openURL(url);
    },
    submitProviderResult: async url => {
      const params = new URLSearchParams(url.split('?')[1]);
      const code = params.get('code');
      const state = params.get('state');
      const providersString = await _Storage.StorageService.get('provider');
      if (providersString) {
        const providers = JSON.parse(providersString);
        const authProvider = providers == null ? void 0 : providers.find(p => p.state === state);
        if (authProvider && code) {
          await (client == null ? void 0 : client.collection('users').authWithOAuth2(authProvider.name, code, authProvider.codeVerifier, typeof document !== 'undefined' ? props.webRedirectUrl : props.mobileRedirectUrl));
        }
      }
    },
    signOut: () => {
      client == null ? void 0 : client.authStore.clear();
    },
    sendPasswordResetEmail: async email => {
      await (client == null ? void 0 : client.collection('users').requestPasswordReset(email));
    },
    sendEmailVerification: async email => {
      await (client == null ? void 0 : client.collection('users').requestVerification(email));
    },
    updateProfile: async (id, record) => {
      await (client == null ? void 0 : client.collection('profiles').update(id, record));
    },
    updateEmail: async email => {
      await (client == null ? void 0 : client.collection('users').requestEmailChange(email));
    },
    deleteUser: async id => {
      await (client == null ? void 0 : client.collection('users').delete(id));
    }
  };
  React.useEffect(() => {
    (async () => {
      const methods = await (client == null ? void 0 : client.collection('users').listAuthMethods());
      setAuthProviders(methods == null ? void 0 : methods.authProviders);
    })();
  }, [props.webRedirectUrl, props.mobileRedirectUrl]);
  return /*#__PURE__*/React.createElement(AuthContext.Provider, {
    value: actions
  }, props.children);
};
exports.AuthProvider = AuthProvider;
